cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/DistributedSchedulerService.java

```diff
@@ -78,6 +82,7 @@ protected void doStop() {
   protected void startUp() throws Exception {
     LOG.info("Starting scheduler.");
     serviceDelegate.startAndWait();
+    countDownLatch.await();
```

Comment:
We also made the change here https://github.com/caskdata/cdap/pull/9076/files#diff-024664c5526c921c18a5f120e7e4edd1R154 to start CoreSchedulerService asynchronously, so that this won't block App-Fabric